### PR TITLE
LRQA-74391 | master

### PR DESCRIPTION
**Pull request by lilylcl**
https://issues.liferay.com/browse/LRQA-74391

### CODE REVIEW

**Comment by lilylcl** [comment]
ci:test:sf

**Comment by Sharryxue** [comment]
Just starting review.

File: modules/apps/portal-language/portal-language-override-test/src/testFunctional/paths/LanguageOverride.path [hunk 1]
```diff
@@ -16,6 +16,19 @@
 	<td></td>
 </tr>
 
+<!--MENU-->
+
+<tr>
+	<td>OPTIONS_ELLIPSIS</td>
```
Comment (Sharryxue) [inline modules/apps/portal-language/portal-language-override-test/src/testFunctional/paths/LanguageOverride.path]: The locator should be duplicate.

File: modules/apps/portal-language/portal-language-override-test/src/testFunctional/paths/LanguageOverride.path [hunk 2]
```diff
@@ -16,6 +16,19 @@
...
+	<td>//div[@class='control-menu-icon']//*[contains(@class,'dropdown-toggle')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>EXPORT_OVERRIDDEN_TRANSLATIONS_OPTION</td>
```
Comment (Sharryxue) [inline modules/apps/portal-language/portal-language-override-test/src/testFunctional/paths/LanguageOverride.path]: Same as this one.

File: modules/apps/portal-language/portal-language-override-test/src/testFunctional/macros/LanguageOverride.macro [hunk 3]
```diff
@@ -52,6 +52,10 @@ definition {
 		Navigator.openWithAppendToBaseURL(urlAppend = "group/control_panel/manage?p_p_id=com_liferay_portal_language_override_web_internal_portlet_PLOPortlet");
 	}
 
+	macro openOptionsMenu {
```
Comment (Sharryxue) [inline modules/apps/portal-language/portal-language-override-test/src/testFunctional/macros/LanguageOverride.macro]: There should be an available macro for use.

File: modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+		if ("${testPortalInstance}" == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@description = "Export option displays when we have language key with override."
+	@priority = "3"
+	test CanViewExportOptionWhenHasOverride {
+		LanguageOverride.addLanguageKey(
```
Comment (Sharryxue) [inline modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase]: Apply the BDD mode to tests.

**Review by Sharryxue — CHANGES_REQUESTED** [review]
Hello @lilylcl,
Good job for starting the NAT, there are things need changes, please refer to the comments. Thanks.

**Comment by lilylcl** [comment]
All done and push again, thanks @Sharryxue for review!

**Comment by lilylcl** [comment]
ci:test:sf

**Comment by Sharryxue** [comment]
Just starting review.

File: modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+		if ("${testPortalInstance}" == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@description = "Export option displays when we have language key with override."
+	@priority = "3"
+	test CanViewExportOptionWhenHasOverride {
+		task ("When a new language keys is created") {
```
Comment (Sharryxue) [inline modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase:27]: The BDD mode start with Given, and goes to When, end with Then.

File: modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		if ("${testPortalInstance}" == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@description = "Export option displays when we have language key with override."
+	@priority = "3"
+	test CanViewExportOptionWhenHasOverride {
```
Comment (Sharryxue) [inline modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase:26]: Please add a step to view the Export option not show up by default.

File: modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+		task ("When a new language keys is created") {
+			LanguageOverride.addLanguageKey(
+				languageKey = "new-key",
+				translationOverrideList = "en-US",
+				translationOverrideValueList = "New key");
+		}
+
+		task ("Then Export Overridden Translations should appear in the options menu dropdown") {
+			Portlet.viewPortletOptions(optionsName = "Export Overridden Translations");
```
Comment (Sharryxue) [inline modules/apps/portal-language/portal-language-override-test/src/testFunctional/tests/LanguageOverrideExportImport.testcase:35]: Great job for finding the exist macro.

**Review by Sharryxue — CHANGES_REQUESTED** [review]
Hello @lilylcl ,
Please add a step a view no Export option by default. Thanks.

**Comment by lilylcl** [comment]
Resend [Sharryxue/liferay-portal#436](https://github.com/Sharryxue/liferay-portal/pull/436)

**Comment by lilylcl** [comment]
Resend https://github.com/Sharryxue/liferay-portal/pull/452

**Pull request closed without merging**